backend/hdf5/DataArrayHDF5.cpp

```diff
@@ -131,13 +131,13 @@ vector<double> DataArrayHDF5::polynomCoefficients() const {
 }
 
 
-void DataArrayHDF5::polynomCoefficients(const vector<double> &coefficients) {
+void DataArrayHDF5::polynomCoefficients(const vector<double> &coefficients, bool compression) {
     DataSet ds;
     if (group().hasData("polynom_coefficients")) {
         ds = group().openData("polynom_coefficients");
         ds.setExtent({coefficients.size()});
     } else {
-        ds = group().createData("polynom_coefficients", H5T_NATIVE_DOUBLE, {coefficients.size()});
+        ds = group().createData("polynom_coefficients", H5T_NATIVE_DOUBLE, {coefficients.size()}, {}, {}, true, true, compression);
```

Comment:
We could (should?) re-order the arguments so we can make use of the default args most of the time. I am not sure about the rest of the codebase but I see a few new `{}, {}, true, true, compression` places. We could avoid that by moving the compression arg up behind size, but maybe then other places need more default args, not sure. Do you have an idea @jgrewe ? The `createData` is internal API so I think we can do that without too much hassle, if we wanted to.